File: tests/Comments.pas
namespace Demo;
// comment after namespace

interface

uses System;

type
  Commented = public class
  public
    procedure SayHello();
  end;

  Foo = class
  public
    method ValueParen: Integer;
    {$REGION 'extra'}
    method ValueBrace: Integer;
    {$ENDREGION}
  method ValueCStyle: Integer;
  method HeaderLineComment;
  end;

  /// <remarks>
  /// comment before class
  /// </remarks>
  PreComment = class
  end;

implementation

procedure Commented.SayHello();
begin
  { region Hello }
  System.Console.WriteLine('Hello');
  { endregion }
end;

method Foo.ValueParen: Integer;
begin
  (* comment with *** inside
     multiple lines *)
  result := 123;
end;

method Foo.ValueBrace: Integer;
begin
  { comment
    multiple lines }
  result := 123;
end;

method Foo.ValueCStyle: Integer;
begin
  /* comment
     multiple lines */
  result := 123;
end;

method Foo.HeaderLineComment; // header comment
var
  i: Integer;
begin
  i := 1;
end;

end.

File: tests/Comments.cs
using System;

namespace Demo {
    public partial class Commented {
        public void SayHello() {
            #region Hello
            System.Console.WriteLine("Hello");
            #endregion
        }
    }
    
    public partial class Foo {
        public int ValueParen() {
            int result;
            /* comment with *** inside
                 multiple lines */
            result = 123;
            return result;
        }
        #region extra
        public int ValueBrace() {
            int result;
            /* comment
                multiple lines */
            result = 123;
            return result;
        }
        #endregion
        public int ValueCStyle() {
            int result;
            /* comment
                 multiple lines */
            result = 123;
            return result;
        }
        public void HeaderLineComment() {
            /* header comment */
            int i;
            i = 1;
        }
    }
    
    /// <remarks>
    /// comment before class
    /// </remarks>
    public partial class PreComment {
    }
}
